staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -728,6 +728,78 @@ type VirtualMachineInstanceGuestOSInfo struct {
 	ID string `json:"id,omitempty"`
 }
 
+// +k8s:openapi-gen=true
+type MigrationNetworkType string
+
+// +k8s:openapi-gen=true
+const (
+	// PodNetworkMigration indicates the migration will happen over the pod network
+	Pod MigrationNetworkType = "Pod"
+	// MigrationNetworkMigration indicates the migration will happen over a dedicated migration network
+	Migration MigrationNetworkType = "Migration"
+)
+
+// +k8s:openapi-gen=true
+type VirtualMachineInstanceMigrationSourceState struct {
+	// The source node that the VMI originated on
+	Node string `json:"node,omitempty"`
+	Pod  string `json:"pod,omitempty"`
+	// The Source VirtualMachineInstanceMigration object associated with this migration
+	MigrationUID types.UID `json:"migrationUID,omitempty"`
+	// If the VMI being migrated uses persistent features (backend-storage), its source PVC name is saved here
+	PersistentStatePVCName string `json:"persistentStatePVCName,omitempty"`
+
+	// Node selectors needed by the target to start the receiving pod.
+	NodeSelectors map[string]string `json:"nodeSelectors,omitempty"`
+
+	// The name of the domain on the source libvirt domain
+	DomainName *string `json:"domainName,omitempty"`
+
+	// Namespace used in the name of the source libvirt domain. Can be used to find and modify paths in the domain
+	DomainNamespace *string `json:"domainNamespace,omitempty"`
+
+	// The URL the synchronization controller can connect to synchronize the VMI migration state
+	SynchronizationConnectionURL *string `json:"synchronizationConnectionURL,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type VirtualMachineInstanceMigrationTargetState struct {
+	// The timestamp at which the target node detects the domain is active
+	NodeDomainReadyTimestamp *metav1.Time `json:"nodeDomainReadyTimestamp,omitempty"`
+	// The Target Node has seen the Domain Start Event
+	NodeDomainDetected bool `json:"nodeDomainDetected,omitempty"`
+	// The address of the target node to use for the migration
+	NodeAddress string `json:"nodeAddress,omitempty"`
+	// The list of ports opened for live migration on the destination node
+	DirectMigrationNodePorts map[string]int `json:"directMigrationNodePorts,omitempty"`
+	// The target node that the VMI is moving to
+	Node string `json:"node,omitempty"`
+	// The target pod that the VMI is moving to
+	Pod string `json:"pod,omitempty"`
+	// The UID of the target attachment pod for hotplug volumes
+	AttachmentPodUID types.UID `json:"attachmentPodUID,omitempty"`
+	// The Target VirtualMachineInstanceMigration object associated with this migration
+	MigrationUID types.UID `json:"migrationUID,omitempty"`
```

Comment:
Nit: maybe we could start the structs with the fields that exist in both source and target? Like Node, Pod, MigrationUID. I think that would make this cleared. They could even be in a shared inlined struct!

Reply:
I get what you mean, the bot suggested it as well. I am just a little hesitant in case we diverge these fields at some time in the future. I guess we can do it now, and move fields out of the common one if they diverge. It feels kind of forced.